src/common/types/types.cpp

```diff
@@ -20,6 +20,101 @@ using kuzu::function::BuiltInFunctionsUtils;
 namespace kuzu {
 namespace common {
 
+LogicalType* ListType::getChildType(const kuzu::common::LogicalType* type) {
+    KU_ASSERT(type->getPhysicalType() == PhysicalTypeID::LIST ||
+              type->getPhysicalType() == PhysicalTypeID::ARRAY);
+    auto listTypeInfo = type->extraTypeInfo->constPtrCast<ListTypeInfo>();
+    return listTypeInfo->getChildType();
+}
+
+LogicalType* ArrayType::getChildType(const LogicalType* type) {
```

Comment:
Can u open an issue for one of our interns to change parameter to const& instead of const*. I don't know why we did const* back in the day.